src/map/GMap.js

```diff
@@ -455,6 +649,66 @@
     GMap.prototype.zoomToFit = function () {
         return this.zoomTo(this.data());
     };
+    
+    GMap.prototype.drawingOptions = function(_) {
+        if(typeof(_) === "object") {
+            if (typeof(this._drawingManager) === "object") {
+                this._drawingManager.setOptions(_);
+            }
+        }
+        return this._drawingManager;
+    };
+    
+    GMap.prototype.userShapeSelection = function (_) {
+        if (!arguments.length) return this._userShapeSelection;
+        if (this._userShapeSelection) {
+            this._userShapeSelection.setEditable(false);
+        }
+        this._userShapeSelection = _;
+        if (this._userShapeSelection) {
+            this._userShapeSelection.setEditable(true);
+        }
+    };
+    
+    GMap.prototype.deleteUserShape = function (_) {
+        if (this._userShapeSelection === _) {
+            this.userShapeSelection(null);
+        }
+        this._userShapes.remove(_);
+    };
+    
+    GMap.prototype.onDrawingComplete = function(event) {
+        if (event.type !== google.maps.drawing.OverlayType.MARKER) {
+            this._drawingManager.setDrawingMode(null);
+            var newShape = event.overlay;
+            newShape.__hpcc_type = event.type;
+            this._userShapes.add(newShape);
+            var context = this;
+            var ctrl = false;
+            window.addEventListener('keydown', function(e) {
+                if (e.keyIdentifier === "Control" || e.ctrlKey === true) {
+                    ctrl = true;
+                }
+            });
+            window.addEventListener('keyup', function(e) {
+                if (e.ctrlKey === false) {
+                    ctrl = false;
+                }
+            });
+            google.maps.event.addListener(newShape, 'click', function (ev) {
+                context.userShapeSelection(newShape);
+                if (ev && ctrl === true) {
```

Comment:
To my understanding there's no guarantee that the "ev" parameter will contain the ctrl key state, that's the reasoning behind having the keydown and keyup handlers.